### ISSUE TITLE
Tensor Object not moved to GPU during Temperature sensitive decoding leading to much higher latency

### ISSUE THREAD

**Issue by anshmn** [guidance-ai/guidance#615]
In the guidance/models/_models.py  , 

```python
if current_temp == 0:
                    sampling_order = np.argsort(-logits) # we need numpy so the enumerate below does not get really slow...
                else:
                    # assert top_p == 1, "Still need to add support for top_p!"
                    if torch:
                        logits = torch.tensor(logits)   ###    ->   logits = torch.tensor(logits).cuda() 
                        torch.div(logits, current_temp, out=logits)
                        probs_torch = torch.nn.functional.softmax(logits, dim=-1)
                        sampling_order = torch.multinomial(probs_torch, len(probs_torch)).cpu().numpy()
```
the logits tensor should be moved to gpu if it is available

Kindly correct  me if this will cause some additional issues

**Comment by prats0599** [comment]
Yeah I believe that the tensor should be moved to whatever device is available.
